File: 3223-minimum-length-of-string-after-operations/3223-minimum-length-of-string-after-operations.java
class Solution {
    public int minimumLength(String s) {
        // count freq of each char
        var freq = new int[26];
        for(var ch : s.toCharArray())
            freq[ch - 'a']++;
        
        // try to reduce all freq to 1 (for odd) & 2 (for even)
        var removed = 0;
        for(var num : freq) {
            if(num < 3) continue;  // can't reduce it
            if((num & 1) == 1) removed += num - 1;  // odd freq
            else removed += num - 2;  // even freq
        }
        
        return s.length() - removed;
    }
}

File: 3223-minimum-length-of-string-after-operations/3223-minimum-length-of-string-after-operations.py
class Solution:
    def minimumLength(self, s: str) -> int:
        # count freq of each char
        freq = [0] * 26
        for ch in s:
            freq[ord(ch) - ord('a')] += 1
        
        # try to reduce all freq to 1 (for odd) & 2 (for even)
        removed = 0
        for num in freq:
            if (num < 3): continue  # can't reduce it
            if (num & 1) == 1: removed += num - 1    # odd freq
            else: removed += num - 2     # even freq
        
        return len(s) - removed
